saleor/invoice/models.py

```diff
@@ -0,0 +1,82 @@
+from django.conf import settings
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+from django.utils.timezone import now
+
+from ..core import JobStatus
+from ..core.models import Job, ModelWithMetadata
+from ..core.utils import build_absolute_uri
+from ..core.utils.json_serializer import CustomJsonEncoder
+from ..graphql.invoice.enums import PendingTarget
+from ..order.models import Order
+from . import InvoiceEvents
+
+
+class InvoiceQueryset(models.QuerySet):
+    def ready(self):
+        return self.filter(job__status=JobStatus.SUCCESS)
+
+
+class Invoice(ModelWithMetadata):
+    order = models.ForeignKey(
+        Order, related_name="invoices", null=True, on_delete=models.SET_NULL
+    )
+    number = models.CharField(max_length=255, null=True)
+    created = models.DateTimeField(null=True)
+    external_url = models.URLField(null=True, max_length=2048)
+    invoice_file = models.FileField(upload_to="invoices")
+    objects = InvoiceQueryset.as_manager()
+
+    @property
+    def url(self):
+        if self.invoice_file:
+            return build_absolute_uri(self.invoice_file.url)
+        return self.external_url
+
+    @url.setter
+    def url(self, value):
+        self.external_url = value
+
+    def update_invoice(self, number=None, url=None):
+        if number is not None:
+            self.number = number
+        if url is not None:
+            self.url = url
+        self.save()
+
+
+class InvoiceJob(Job):
+    invoice = models.OneToOneField(
+        Invoice, on_delete=models.CASCADE, related_name="job"
+    )
+    pending_target = models.CharField(max_length=50, choices=PendingTarget.CHOICES)
```

Comment:
Could you explain what is the reason for using the additional status field? Why Job's `status` is not enough?

Reply:
Added a docstring. When an Invoice is pending it can transform to DELTED or SUCCESS (we don't have "PENDING_DELETE" status on Job), this field holds that information for extra logic in plugin.

Reply:
Do we need a pending delete status (is it going to be used for anything)? If so, we can add it to the Job interface.

Reply:
DELETING invoice is such a long action that we need special status for that?

Reply:
@IKarbowiak It can be done async through external service.